File: 121. Best Time to Buy and Sell Stock/main.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        max_profit=0

        highest=prices[-1]


        for i in range(len(prices)-1,-1,-1):
            max_profit=max(max_profit,highest-prices[i])
            highest=max(highest,prices[i])


        return max_profit

        

File: 121. Best Time to Buy and Sell Stock/main.cpp
#include <bits/stdc++.h>



class Solution {
public:
    int maxProfit(std::vector<int>& prices) {

        int maxProfit=0;
        int n=prices.size();

        int highest=prices[n-1];
        for(int i=n-1;i>=0;i--)
        {
            maxProfit=std::max(highest-prices[i],maxProfit);
            highest=std::max(prices[i],highest);
            
        }


        return maxProfit;
        
    }
};